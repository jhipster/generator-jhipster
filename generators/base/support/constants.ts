export const LOCAL_BLUEPRINT_PACKAGE_NAMESPACE = '@jhipster/generator-jhipster-local';
