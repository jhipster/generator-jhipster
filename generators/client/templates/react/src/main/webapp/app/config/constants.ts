const config = {
  version: process.env.version || '0.0.0'
};

export default config;

export const messages = {
  DATA_CREATE_SUCCESS_ALERT: 'Data created successfully!',
  DATA_DELETE_SUCCESS_ALERT: 'Data deleted successfully!',
  DATA_UPDATE_ERROR_ALERT: 'Data deleted successfully!'
};
