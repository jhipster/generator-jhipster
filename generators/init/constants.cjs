/**
 * Copyright 2013-2022 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const NODE_VERSION = '14.18.1';

const PRETTIER_DEFAULT_INDENT = 'prettierDefaultIndent';
const PRETTIER_DEFAULT_INDENT_DEFAULT_VALUE = 2;

const SKIP_COMMIT_HOOK = 'skipCommitHook';
const SKIP_COMMIT_HOOK_DESCRIPTION = 'Skip adding husky commit hooks';
const SKIP_COMMIT_HOOK_DEFAULT_VALUE = false;

module.exports = {
  NODE_VERSION,
  PRETTIER_DEFAULT_INDENT,
  PRETTIER_DEFAULT_INDENT_DEFAULT_VALUE,
  SKIP_COMMIT_HOOK,
  SKIP_COMMIT_HOOK_DESCRIPTION,
  SKIP_COMMIT_HOOK_DEFAULT_VALUE,
};
