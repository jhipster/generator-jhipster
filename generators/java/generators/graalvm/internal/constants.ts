// renovate: datasource=github-releases depName=graalvm-reachability-metadata oracle/graalvm-reachability-metadata
export const GRAALVM_REACHABILITY_METADATA = '0.3.14';
