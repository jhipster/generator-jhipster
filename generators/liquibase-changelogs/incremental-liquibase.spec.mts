import path, { basename, join } from 'path';
import { jestExpect as expect } from 'mocha-expect-snapshot';
import { existsSync, mkdirSync, writeFileSync } from 'fs';
import { fileURLToPath } from 'url';

import { skipPrettierHelpers as helpers } from '../../test/support/helpers.mjs';
import constants from '../generator-constants.cjs';
import { jdlImporter } from '../../jdl/index.mjs';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const { createImporterFromContent } = jdlImporter;
const { SERVER_MAIN_RES_DIR } = constants;

const incrementalFiles = [
  `${SERVER_MAIN_RES_DIR}config/liquibase/master.xml`,
  `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/00000000000000_initial_schema.xml`,
];

const DEFAULT_TEST_OPTIONS = { skipInstall: true, skipChecks: true, skipPrettier: true };

const baseName = 'JhipsterApp';

const jdlApplication = `
application {
    config { baseName ${baseName} }
    entities *
}`;

const jdlApplicationWithEntities = `
${jdlApplication}
entity One {
    original String
}
entity Another {
    original String
}`;

const jdlApplicationWithRelationshipToUser = `
${jdlApplicationWithEntities}
relationship ManyToOne {
    One{user(login)} to User
}
`;

const jdlApplicationEntitieWithByteTypes = `
${jdlApplication}
entity Smarty {
  name String required unique minlength(2) maxlength(10)
  price Float required min(0)
  description TextBlob required
  picture ImageBlob required
  specification Blob
  category ProductCategory
  inventory Integer required min(0)
}
enum ProductCategory {
  Laptop, Desktop, Phone, Tablet, Accessory
}`;

const jdlApplicationEntitieWithoutByteTypes = `
${jdlApplication}
entity Smarty {
  name String
  age Integer
  height Long
  income BigDecimal
  expense Double
  savings Float
  category ProductCategory
  happy Boolean
  dob LocalDate
  exactTime ZonedDateTime
  travelTime Duration
  moment Instant
}
enum ProductCategory {
  Laptop, Desktop, Phone, Tablet, Accessory
}`;

const generatorPath = join(__dirname, '../server/index.mjs');
const mockedGenerators = ['jhipster:common', 'jhipster:gradle', 'jhipster:maven'];

describe('jhipster:app --incremental-changelog', function () {
  this.timeout(45000);
  const options = {
    ...DEFAULT_TEST_OPTIONS,
    noInsight: true,
    creationTimestamp: '2020-01-01',
    incrementalChangelog: true,
    defaults: true,
    skipClient: true,
    force: true,
    withEntities: true,
  };
  context('when creating a new application', () => {
    let runResult;
    before(async () => {
      runResult = await helpers.run(generatorPath).withOptions(options).withMockedGenerators(mockedGenerators);
    });

    after(() => runResult.cleanup());

    it('should create application', () => {
      runResult.assertFile(['.yo-rc.json']);
    });

    it('should match snapshot', () => {
      expect(runResult.getSnapshot('**/src/main/resources/config/liquibase/**')).toMatchSnapshot();
    });
  });

  context('when incremental liquibase files exists', () => {
    context('with default options', () => {
      let runResult;
      before(async () => {
        runResult = await helpers
          .create(generatorPath)
          .withOptions(options)
          .doInDir(cwd => {
            incrementalFiles.forEach(filePath => {
              filePath = join(cwd, filePath);
              const dirname = path.dirname(filePath);
              if (!existsSync(dirname)) {
                mkdirSync(dirname, { recursive: true });
              }
              writeFileSync(filePath, basename(filePath));
            });
          })
          .run();
      });

      after(() => runResult.cleanup());

      it('should create application', () => {
        runResult.assertFile(['.yo-rc.json']);
      });

      it('should not override existing incremental files', () => {
        incrementalFiles.forEach(filePath => {
          runResult.assertFileContent(filePath, basename(filePath));
        });
      });

      it('should match snapshot', () => {
        expect(runResult.getSnapshot('**/src/main/resources/config/liquibase/**')).toMatchSnapshot();
      });
    });

    context('with --recreate-initial-changelog', () => {
      let runResult;
      before(async () => {
        runResult = await helpers
          .create(generatorPath)
          .withOptions({ ...options, recreateInitialChangelog: true })
          .doInDir(cwd => {
            incrementalFiles.forEach(filePath => {
              filePath = join(cwd, filePath);
              const dirname = path.dirname(filePath);
              if (!existsSync(dirname)) {
                mkdirSync(dirname, { recursive: true });
              }
              writeFileSync(filePath, basename(filePath));
            });
          })
          .run();
      });

      after(() => runResult.cleanup());

      it('should create application', () => {
        runResult.assertFile(['.yo-rc.json']);
      });

      it('should override existing incremental files', () => {
        incrementalFiles.forEach(filePath => {
          runResult.assertNoFileContent(filePath, filePath);
        });
      });

      it('should match snapshot', () => {
        expect(runResult.getSnapshot('**/src/main/resources/config/liquibase/**')).toMatchSnapshot();
      });
    });
  });

  context('regenerating the application', () => {
    let runResult;
    before(async () => {
      const initialState = createImporterFromContent(jdlApplicationWithRelationshipToUser, {
        ...options,
        skipFileGeneration: true,
        creationTimestampConfig: options.creationTimestamp,
      }).import();
      const applicationWithEntities = initialState.exportedApplicationsWithEntities[baseName];
      expect(applicationWithEntities).toBeTruthy();
      expect(applicationWithEntities.entities.length).toBe(2);
      runResult = await helpers
        .create(generatorPath)
        .withOptions({ ...options, applicationWithEntities })
        .run();
      const state = createImporterFromContent(jdlApplicationWithRelationshipToUser, {
        skipFileGeneration: true,
        ...options,
      }).import();
      runResult = await runResult
        .create(generatorPath)
        .withOptions({
          ...options,
          applicationWithEntities: state.exportedApplicationsWithEntities.JhipsterApp,
          creationTimestamp: '2020-01-02',
        })
        .run();
    });

    after(() => runResult.cleanup());

    it('should create application', () => {
      runResult.assertFile(['.yo-rc.json']);
    });
    it('should create entity config file', () => {
      runResult.assertFile([join('.jhipster', 'One.json'), join('.jhipster', 'Another.json')]);
    });
    it('should create entity initial changelog', () => {
      runResult.assertFile([
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200101000100_added_entity_One.xml`,
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200101000200_added_entity_Another.xml`,
      ]);
    });
    it('should create entity initial fake data', () => {
      runResult.assertFile([
        `${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000100_entity_one.csv`,
        `${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000200_entity_another.csv`,
      ]);
    });
    it('should not create the entity update changelog', () => {
      runResult.assertNoFile([
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_One.xml`,
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_constraints_One.xml`,
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000200_updated_entity_Another.xml`,
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000200_updated_entity_constraints_Another.xml`,
      ]);
    });

    it('should match snapshot', () => {
      expect(runResult.getSnapshot('**/src/main/resources/config/liquibase/**')).toMatchSnapshot();
    });
  });

  context('when adding a field without constraints', () => {
    let runResult;
    before(async () => {
      const baseName = 'JhipsterApp';
      const initialState = createImporterFromContent(
        `
${jdlApplication}
entity Customer {
    original String
}
`,
        {
          ...options,
          skipFileGeneration: true,
          creationTimestampConfig: options.creationTimestamp,
        }
      ).import();
      const applicationWithEntities = initialState.exportedApplicationsWithEntities[baseName];
      expect(applicationWithEntities).toBeTruthy();
      expect(applicationWithEntities.entities.length).toBe(1);
      runResult = await helpers
        .create(generatorPath)
        .withOptions({ ...options, applicationWithEntities })
        .run();

      const state = createImporterFromContent(
        `
${jdlApplication}
entity Customer {
    original String
    foo String
}
`,
        {
          skipFileGeneration: true,
          ...options,
        }
      ).import();
      runResult = await runResult
        .create(generatorPath)
        .withOptions({
          ...options,
          applicationWithEntities: state.exportedApplicationsWithEntities.JhipsterApp,
          creationTimestamp: '2020-01-02',
        })
        .run();
    });

    after(() => runResult.cleanup());

    it('should create application', () => {
      runResult.assertFile(['.yo-rc.json']);
    });
    it('should create entity config file', () => {
      runResult.assertFile([join('.jhipster', 'Customer.json')]);
    });
    it('should create entity initial changelog', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200101000100_added_entity_Customer.xml`]);
    });
    it('should create entity initial fake data', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000100_entity_customer.csv`]);
    });
    it('should create entity update changelog with addColumn', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`]);
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'addColumn tableName="customer"'
      );
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'column name="foo" type="varchar(255)"'
      );
      runResult.assertNoFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'dropColump'
      );
    });
    it('should not create the entity constraint update changelog', () => {
      runResult.assertNoFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_constraints_Customer.xml`]);
    });
    it('should match snapshot', () => {
      expect(runResult.getSnapshot('**/src/main/resources/config/liquibase/**')).toMatchSnapshot();
    });
  });

  context('when adding a field with constraints', () => {
    let runResult;
    before(async () => {
      const baseName = 'JhipsterApp';
      const initialState = createImporterFromContent(
        `
${jdlApplication}
entity Customer {
    original String
}
`,
        {
          ...options,
          skipFileGeneration: true,
          creationTimestampConfig: options.creationTimestamp,
        }
      ).import();
      const applicationWithEntities = initialState.exportedApplicationsWithEntities[baseName];
      expect(applicationWithEntities).toBeTruthy();
      expect(applicationWithEntities.entities.length).toBe(1);
      runResult = await helpers
        .create(generatorPath)
        .withOptions({ ...options, applicationWithEntities })
        .run();

      const regenerateState = createImporterFromContent(
        `
${jdlApplication}
entity Customer {
  original String
  foo String required
}
`,
        {
          skipFileGeneration: true,
          ...options,
        }
      ).import();

      runResult = await runResult
        .create(generatorPath)
        .withOptions({
          ...options,
          applicationWithEntities: regenerateState.exportedApplicationsWithEntities.JhipsterApp,
          creationTimestamp: '2020-01-02',
        })
        .run();
    });

    after(() => runResult.cleanup());

    it('should create application', () => {
      runResult.assertFile(['.yo-rc.json']);
    });
    it('should create entity config file', () => {
      runResult.assertFile([join('.jhipster', 'Customer.json')]);
    });
    it('should create entity initial changelog', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200101000100_added_entity_Customer.xml`]);
    });
    it('should create entity initial fake data', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000100_entity_customer.csv`]);
    });
    it('should create entity update changelog with addColumn', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`]);
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'addColumn tableName="customer"'
      );
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'column name="foo" type="varchar(255)"'
      );
      runResult.assertNoFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'dropColump'
      );
    });
    it('should create the entity constraint update changelog', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_constraints_Customer.xml`]);
    });
    it('should match snapshot', () => {
      expect(runResult.getSnapshot('**/src/main/resources/config/liquibase/**')).toMatchSnapshot();
    });
  });

  context('when removing a field without constraints', () => {
    let runResult;
    before(async () => {
      const baseName = 'JhipsterApp';
      const initialState = createImporterFromContent(
        `
${jdlApplication}
entity Customer {
    original String
    foo String
}
`,
        {
          ...options,
          skipFileGeneration: true,
          creationTimestampConfig: options.creationTimestamp,
        }
      ).import();
      const applicationWithEntities = initialState.exportedApplicationsWithEntities[baseName];
      expect(applicationWithEntities).toBeTruthy();
      expect(applicationWithEntities.entities.length).toBe(1);
      runResult = await helpers
        .create(generatorPath)
        .withOptions({ ...options, applicationWithEntities })
        .run();

      const state = createImporterFromContent(
        `
${jdlApplication}
entity Customer {
    original String
}
`,
        {
          skipFileGeneration: true,
          ...options,
        }
      ).import();
      runResult = await runResult
        .create(generatorPath)
        .withOptions({
          ...options,
          applicationWithEntities: state.exportedApplicationsWithEntities[baseName],
          creationTimestamp: '2020-01-02',
        })
        .run();
    });

    after(() => runResult.cleanup());

    it('should create application', () => {
      runResult.assertFile(['.yo-rc.json']);
    });
    it('should create entity config file', () => {
      runResult.assertFile([join('.jhipster', 'Customer.json')]);
    });
    it('should create entity initial changelog', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200101000100_added_entity_Customer.xml`]);
    });
    it('should create entity initial fake data', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000100_entity_customer.csv`]);
    });
    it('should create entity update changelog with dropColumn', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`]);
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'dropColumn tableName="customer"'
      );
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'column name="foo"'
      );
      runResult.assertNoFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'addColumn'
      );
    });
    it('should not create the entity constraint update changelog', () => {
      runResult.assertNoFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_constraints_Customer.xml`]);
    });
    it('should match snapshot', () => {
      expect(runResult.getSnapshot('**/src/main/resources/config/liquibase/**')).toMatchSnapshot();
    });
  });

  context('when removing a field with constraints', () => {
    let runResult;
    before(async () => {
      const baseName = 'JhipsterApp';
      const initialState = createImporterFromContent(
        `
${jdlApplication}
entity Customer {
    original String
    foo String required
}
`,
        {
          ...options,
          skipFileGeneration: true,
          creationTimestampConfig: options.creationTimestamp,
        }
      ).import();
      const applicationWithEntities = initialState.exportedApplicationsWithEntities[baseName];
      expect(applicationWithEntities).toBeTruthy();
      expect(applicationWithEntities.entities.length).toBe(1);
      runResult = await helpers
        .create(generatorPath)
        .withOptions({ ...options, applicationWithEntities })
        .run();

      const state = createImporterFromContent(
        `
${jdlApplication}
entity Customer {
    original String
}
`,
        {
          skipFileGeneration: true,
          ...options,
        }
      ).import();
      runResult = await runResult
        .create(generatorPath)
        .withOptions({
          ...options,
          applicationWithEntities: state.exportedApplicationsWithEntities[baseName],
          creationTimestamp: '2020-01-02',
        })
        .run();
    });

    after(() => runResult.cleanup());

    it('should create application', () => {
      runResult.assertFile(['.yo-rc.json']);
    });
    it('should create entity config file', () => {
      runResult.assertFile([join('.jhipster', 'Customer.json')]);
    });
    it('should create entity initial changelog', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200101000100_added_entity_Customer.xml`]);
    });
    it('should create entity initial fake data', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000100_entity_customer.csv`]);
    });
    it('should create entity update changelog with dropColumn', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`]);
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'dropColumn tableName="customer"'
      );
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'column name="foo"'
      );
      runResult.assertNoFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_Customer.xml`,
        'addColumn'
      );
    });
    it('should create the entity constraint update changelog', () => {
      runResult.assertNoFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_constraints_Customer.xml`]);
    });
    it('should match snapshot', () => {
      expect(runResult.getSnapshot('**/src/main/resources/config/liquibase/**')).toMatchSnapshot();
    });
  });

  context('when adding a relationship', () => {
    let runResult;
    before(async () => {
      const baseName = 'JhipsterApp';
      const initialState = createImporterFromContent(jdlApplicationWithEntities, {
        ...options,
        skipFileGeneration: true,
        creationTimestampConfig: options.creationTimestamp,
      }).import();
      const applicationWithEntities = initialState.exportedApplicationsWithEntities[baseName];
      expect(applicationWithEntities).toBeTruthy();
      expect(applicationWithEntities.entities.length).toBe(2);
      runResult = await helpers
        .create(generatorPath)
        .withOptions({ ...options, applicationWithEntities })
        .run();

      const state = createImporterFromContent(
        `
${jdlApplicationWithEntities}
relationship OneToOne {
    One to Another,
}
`,
        {
          skipFileGeneration: true,
          ...options,
        }
      ).import();
      runResult = await runResult
        .create(generatorPath)
        .withOptions({
          ...options,
          applicationWithEntities: state.exportedApplicationsWithEntities[baseName],
          creationTimestamp: '2020-01-02',
        })
        .run();
    });

    after(() => runResult.cleanup());

    it('should create application', () => {
      runResult.assertFile(['.yo-rc.json']);
    });
    it('should create entity config file', () => {
      runResult.assertFile([join('.jhipster', 'One.json'), join('.jhipster', 'Another.json')]);
    });
    it('should create entity initial changelog', () => {
      runResult.assertFile([
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200101000100_added_entity_One.xml`,
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200101000200_added_entity_Another.xml`,
      ]);
    });
    it('should create entity initial fake data', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000100_entity_one.csv`]);
    });
    it('should create entity update changelog with addColumn', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_One.xml`]);
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_One.xml`,
        'addColumn tableName="one"'
      );
      runResult.assertFileContent(
        `${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_One.xml`,
        'column name="another_id" type="bigint"'
      );
      runResult.assertNoFileContent(`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_One.xml`, 'dropColumn');
    });
    it('should create the entity constraint update changelog', () => {
      runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/changelog/20200102000100_updated_entity_constraints_One.xml`]);
    });
    it('should match snapshot', () => {
      expect(runResult.getSnapshot('**/src/main/resources/config/liquibase/**')).toMatchSnapshot();
    });
  });
  context('entities with/without byte fields should create fake data', () => {
    [
      {
        entity: jdlApplicationEntitieWithByteTypes,
        bytesFields: true,
        testContent:
          '1;geez;4178;../fake-data/blob/hipster.txt;../fake-data/blob/hipster.png;image/png;../fake-data/blob/hipster.png;image/png;Laptop;17242',
        contentRequired: true,
      },
      {
        entity: jdlApplicationEntitieWithoutByteTypes,
        bytesFields: false,
        testContent: 'content_type',
        contentRequired: false,
      },
    ].forEach(eachEntityConfig => {
      describe(`testing ${eachEntityConfig.bytesFields ? 'with' : 'without'} byte fields`, () => {
        let runResult;
        before(async () => {
          const baseName = 'JhipsterApp';
          const initialState = createImporterFromContent(eachEntityConfig.entity, {
            ...options,
            skipFileGeneration: true,
            creationTimestampConfig: options.creationTimestamp,
          }).import();
          const applicationWithEntities = initialState.exportedApplicationsWithEntities[baseName];
          expect(applicationWithEntities).toBeTruthy();
          expect(applicationWithEntities.entities.length).toBe(1);
          runResult = await helpers
            .create(generatorPath)
            .withOptions({ ...options, applicationWithEntities })
            .run();
        });

        after(() => runResult.cleanup());

        it('should create entity config file', () => {
          runResult.assertFile([join('.jhipster', 'Smarty.json')]);
        });
        it('should create entity initial fake data file', () => {
          runResult.assertFile([`${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000100_entity_smarty.csv`]);
        });
        it('should create fake data file with required content', () => {
          eachEntityConfig.contentRequired
            ? runResult.assertFileContent(
                `${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000100_entity_smarty.csv`,
                eachEntityConfig.testContent
              )
            : runResult.assertNoFileContent(
                `${SERVER_MAIN_RES_DIR}config/liquibase/fake-data/20200101000100_entity_smarty.csv`,
                eachEntityConfig.testContent
              );
        });
      });
    });
  });
});
