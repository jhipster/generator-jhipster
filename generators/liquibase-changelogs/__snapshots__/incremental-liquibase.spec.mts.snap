// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator - app - --incremental-changelog when creating a new application should match snapshot 1`] = `
{
  "src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml": {
    "contents": "<?xml version="1.0" encoding="utf-8"?>
<databaseChangeLog
    xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
    xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd
                        http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">

    <changeSet id="00000000000000" author="jhipster">
        <createSequence sequenceName="sequence_generator" startValue="1050" incrementBy="50"/>
    </changeSet>

    <!--
        JHipster core tables.
        The initial schema has the '00000000000001' id, so that it is over-written if we re-generate it.
    -->
    <changeSet id="00000000000001" author="jhipster">
        <createTable tableName="jhi_user">
            <column name="id" type="bigint">
                <constraints primaryKey="true" nullable="false"/>
            </column>
            <column name="login" type="varchar(50)">
                <constraints unique="true" nullable="false" uniqueConstraintName="ux_user_login"/>
            </column>
            <column name="password_hash" type="varchar(60)"/>
            <column name="first_name" type="varchar(50)"/>
            <column name="last_name" type="varchar(50)"/>
            <column name="email" type="varchar(191)">
                <constraints unique="true" nullable="true" uniqueConstraintName="ux_user_email"/>
            </column>
            <column name="image_url" type="varchar(256)"/>
            <column name="activated" type="boolean" valueBoolean="false">
                <constraints nullable="false" />
            </column>
            <column name="lang_key" type="varchar(10)"/>
            <column name="activation_key" type="varchar(20)"/>
            <column name="reset_key" type="varchar(20)"/>
            <column name="created_by" type="varchar(50)">
                <constraints nullable="false"/>
            </column>
            <column name="created_date" type="timestamp"/>
            <column name="reset_date" type="timestamp">
                <constraints nullable="true"/>
            </column>
            <column name="last_modified_by" type="varchar(50)"/>
            <column name="last_modified_date" type="timestamp"/>
        </createTable>

        <createTable tableName="jhi_authority">
            <column name="name" type="varchar(50)">
                <constraints primaryKey="true" nullable="false"/>
            </column>
        </createTable>

        <createTable tableName="jhi_user_authority">
            <column name="user_id" type="bigint">
                <constraints nullable="false"/>
            </column>
            <column name="authority_name" type="varchar(50)">
                <constraints nullable="false"/>
            </column>
        </createTable>

        <addPrimaryKey columnNames="user_id, authority_name" tableName="jhi_user_authority"/>

        <addForeignKeyConstraint baseColumnNames="authority_name"
                                 baseTableName="jhi_user_authority"
                                 constraintName="fk_authority_name"
                                 referencedColumnNames="name"
                                 referencedTableName="jhi_authority"/>

        <addForeignKeyConstraint baseColumnNames="user_id"
                                 baseTableName="jhi_user_authority"
                                 constraintName="fk_user_id"
                                 referencedColumnNames="id"
                                 referencedTableName="jhi_user"/>

        <addNotNullConstraint   columnName="password_hash"
                                columnDataType="varchar(60)"
                                tableName="jhi_user"/>
        <loadData
                  file="config/liquibase/data/authority.csv"
                  separator=";"
                  tableName="jhi_authority"
                  usePreparedStatements="true">
            <column name="name" type="string"/>
        </loadData>
        <loadData
                  file="config/liquibase/data/user.csv"
                  separator=";"
                  tableName="jhi_user"
                  usePreparedStatements="true">
            <column name="id" type="numeric"/>
            <column name="activated" type="boolean"/>
            <column name="created_date" type="timestamp"/>
        </loadData>
        <loadData
                  file="config/liquibase/data/user_authority.csv"
                  separator=";"
                  tableName="jhi_user_authority"
                  usePreparedStatements="true">
            <column name="user_id" type="numeric"/>
        </loadData>
        <dropDefaultValue tableName="jhi_user" columnName="created_date" columnDataType="\${datetimeType}"/>
    </changeSet>

    <changeSet author="jhipster" id="00000000000002" context="test">
        <createTable tableName="jhi_date_time_wrapper">
            <column  name="id" type="BIGINT">
                <constraints primaryKey="true" primaryKeyName="jhi_date_time_wrapperPK"/>
            </column>
            <column name="instant" type="timestamp"/>
            <column name="local_date_time" type="timestamp"/>
            <column name="offset_date_time" type="timestamp"/>
            <column name="zoned_date_time" type="timestamp"/>
            <column name="local_time" type="time"/>
            <column name="offset_time" type="time"/>
            <column name="local_date" type="date"/>
        </createTable>
    </changeSet>
</databaseChangeLog>
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/liquibase/data/authority.csv": {
    "contents": "name
ROLE_ADMIN
ROLE_USER
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/liquibase/data/user.csv": {
    "contents": "id;login;password_hash;first_name;last_name;email;image_url;activated;lang_key;created_by;last_modified_by
1;admin;$2a$10$gSAhZrxMllrbgj/kkK9UceBPpChGWJA7SYIb1Mqo.n5aNLq1/oRrC;Administrator;Administrator;admin@localhost;;true;en;system;system
2;user;$2a$10$VEjxo0jq2YG9Rbk2HmX9S.k1uZBGYUHdUcid3g/vfiEl7lwWgOH/K;User;User;user@localhost;;true;en;system;system
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/liquibase/data/user_authority.csv": {
    "contents": "user_id;authority_name
1;ROLE_ADMIN
1;ROLE_USER
2;ROLE_USER
",
    "stateCleared": "modified",
  },
  "src/main/resources/config/liquibase/master.xml": {
    "contents": "<?xml version="1.0" encoding="utf-8"?>
<databaseChangeLog
    xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
    <property name="now" value="now()" dbms="h2"/>
    <property name="floatType" value="float4" dbms="h2"/>
    <property name="uuidType" value="uuid" dbms="h2"/>
    <property name="datetimeType" value="datetime" dbms="h2"/>
    <property name="clobType" value="longvarchar" dbms="h2"/>
    <property name="blobType" value="blob" dbms="h2"/>
    <property name="now" value="current_timestamp" dbms="postgresql"/>
    <property name="floatType" value="float4" dbms="postgresql"/>
    <property name="clobType" value="clob" dbms="postgresql"/>
    <property name="blobType" value="blob" dbms="postgresql"/>
    <property name="uuidType" value="uuid" dbms="postgresql"/>
    <property name="datetimeType" value="datetime" dbms="postgresql"/>

    <include file="config/liquibase/changelog/00000000000000_initial_schema.xml" relativeToChangelogFile="false"/>
    <!-- jhipster-needle-liquibase-add-changelog - JHipster will add liquibase changelogs here -->
    <!-- jhipster-needle-liquibase-add-constraints-changelog - JHipster will add liquibase constraints changelogs here -->
    <!-- jhipster-needle-liquibase-add-incremental-changelog - JHipster will add incremental liquibase changelogs here -->
</databaseChangeLog>
",
    "stateCleared": "modified",
  },
}
`;
